### ISSUE TITLE
When using formio/react inside a component, every time the parent component change, formio/react is destroyed and recreated.

### ISSUE THREAD

**Issue by e-gauthier** [formio/react#385]
## Environment

React-formio version: "5.0.0-rc1"  (Possibly all versions)
Formio.js version: "4.12.7"
Frontend framework: React "16.14.0"
Frontend framework: React-Dom "16.14.0"
Browser: many browsers.
Browser version: many browser versions.

## Context 
Our software need to save the schema of formio in a database along with other information about a specific entity. The schema can also be altered by an editor. Ex: Updating a label text without the need to use formio. So, react-formio is embed inside a component called FormioWrapper. Finally, this component is used inside EditEntityComponent with a save button.
Each time there is a change in EditEntityComponent props/states, formio reset itself.

## Steps to Reproduce
1. Go to [https://codesandbox.io/s/cra-react-formio-forked-esi5m?file=/src/index.js](https://codesandbox.io/s/cra-react-formio-forked-esi5m?file=/src/index.js)
2. Click on Advanced tab to open it.
3. Delete, add or move a text box.

## Observed behavior
Formio reset itself by closing the Advanced tab, the actual state is lost. 

## Expected behavior
Formio don't have to reset itself every time a modification is triggered.

## Where to look
After investigation, we found that this block of code is what cause the situation.

src/components/FormBuilder.jsx
line 56 to 59

```js
  useEffect(() => {
    initializeBuilder(props);
    return () => builder ? builder.instance.destroy(true) : null;
  }, [props.form.display, props.form.components, props.options]);
```
**Why do the component need to destroy itself every time a change is triggered?  Is there a limitation that avoid the component to stop destroying itself?**

**Comment by ccontour** [comment]
This is now associated with a ticket in our system to look into further - FIO-2660

**Comment by e-gauthier** [comment]
@ccontour Is it possible to have an estimated time?

**Comment by ccontour** [comment]
https://github.com/formio/react/pull/389

**Comment by JeromeS4RB** [comment]
Any update on this?

**Comment by wag110894** [comment]
As Cameron Contour provided above, we do have a PR in for this fix. We will let you know once this has been merged and tested and resolved on a stable build.

**Comment by pboucher1960** [comment]
Hi, last comment is from Apr 7 and still no news since that time. Any idea when this PR might get merged and distributed in a stable build? Thank you!

**Comment by wag110894** [comment]
This PR has been merged into React Version 5.1.0-rc.1.
